client/data/followers/use-followers.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { useInfiniteQuery } from 'react-query';
+import { uniqueBy } from '@automattic/js-utils';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'calypso/lib/wp';
+
+const MAX_FOLLOWERS = 10; // means pages (= 1000 followers);
+
+/**
+ * Normalizes a follower object. Changes 'avatar' to 'avatar_URL' allowing a follower
+ * object to be used with the Gravatar component.
+ *
+ * @param  {object} follower A follower ojbect
+ * @returns {object}          A normalized follower object
+ */
+export function normalizeFollower( follower ) {
+	follower.avatar_URL = follower.avatar;
+	return follower;
+}
+
+const extractPages = ( pages = [] ) =>
+	pages.flatMap( ( page ) => page.subscribers ).map( normalizeFollower );
+const compareUnique = ( a, b ) => a.ID === b.ID;
+
+const useFollowers = ( siteId, fetchOptions = {}, queryOptions = {} ) => {
+	const { search, type } = fetchOptions;
```

Comment:
`type` should be a mandatory second arg that defaults to `wpcom`. If the `type` query arg is omitted from query, the REST endpoint will treat it as `wpcom`. And because `type` is used as part of the query key, we should avoid storing two identical responses under two keys, one with `type = wpcom` and other with `type = null`.